src/tests/ftest/deployment/server_rank_failure.py

```diff
@@ -0,0 +1,423 @@
+        self.log.info("Start daos_server and detect the DAOS I/O engine message")
+        self.server_managers[0].restart(hosts=self.hostlist_servers)
+
+    def kill_engine(self, engine_kill_host):
+        """Kill engine on the given host.
+
+        Args:
+            engine_kill_host (str): Hostname to kill engine.
+        """
+        pattern = self.server_managers[0].manager.job.command_regex
+        result = stop_processes(hosts=[engine_kill_host], pattern=pattern)
+        if 0 in result and len(result) == 1:
+            self.log.info("No remote daos_engine process killed!")
+        else:
+            self.log.info("daos_engine in %s killed", engine_kill_host)
+
+    def verify_ior_worked(self, ior_results, job_num, errors):
+        """Verify that the IOR worked.
+
+        Args:
+            ior_results (dict): Dictionary that contains the IOR run results.
+            job_num (int): Job number used for the IOR run.
+            errors (list): Error list used in the test.
+        """
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append("Error found in IOR job {}! {}".format(job_num, ior_error))
+
+    def verify_rank_failure(self, ior_namespace, container_namespace):
+        """Verify engine failure can be recovered by restarting daos_server.
+
+        1. Create a pool and a container. Create a container with or without redundancy
+        factor based on container_namespace.
+        2. Get the amount of IO inflow to each rank. We expect them to be 0.
+        3. Run IOR with given object class (IO experiment run).
+        4. Get the amount of IO inflow again.
+        5. Get the rank with the largest amount of inflow. Find out its hostname. We'll
+        kill engines here. If we use replication, some ranks receive more data than
+        others. e.g., With RP_2G1, two ranks get most of the data.
+        6. Run IOR with given object class (Main test run).
+        7. While IOR is running, kill daos_engine processes from the node to simulate a
+        node failure. (Two engines per node.)
+        8. Verify that IOR failed.
+        9. Restart daos_server service on the two nodes.
+        10. Verify the system status by calling dmg system query.
+        11. Verify that the container Health is HEALTHY.
+        12. Run IOR to the same container and verify that it works.
+
+        Args:
+            ior_namespace (str): Yaml namespace that defines the object class used for
+                IOR.
+            container_namespace (str): Yaml namespace that defines the container
+                redundancy factor.
+        """
+        # 1. Create a pool and a container.
+        self.add_pool(namespace="/run/pool_size_ratio_80/*")
+        self.add_container(pool=self.pool, namespace=container_namespace)
+
+        # 2. Get the amount of IO inflow to each rank.
+        metric = "engine_pool_ops_fetch"
```

Comment:
`engine_pool_ops_update` should be used here. Fetch ops will be updated when we perform read operations. As per the yaml file below the ior is only performing write operation `flags: -k -v -w -W`